File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []  # 결과를 저장할 리스트
    days_left = []  # 각 작업의 완료까지 남은 일자를 저장할 리스트
    
    # 각 작업의 완료까지 남은 일자를 계산하여 days_left 리스트에 저장
    for progress, speed in zip(progresses, speeds):
        days = (100 - progress) // speed
        if (100 - progress) % speed != 0:
            days += 1
        days_left.append(days)
    
    # 각 배포마다 몇 개의 작업이 완료되는지 계산
    count = 1
    max_days = days_left[0]
    for i in range(1, len(days_left)):
        if days_left[i] <= max_days:
            count += 1
        else:
            answer.append(count)
            max_days = days_left[i]
            count = 1
    
    answer.append(count)  # 마지막 배포에 대한 결과 추가
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> days = new ArrayList<>();
        
        for (int i = 0; i < progresses.length; i++) {
            int daysToComplete = (100 - progresses[i] + speeds[i] - 1) / speeds[i]; 
            days.add(daysToComplete);
        }

        ArrayList<Integer> result = new ArrayList<>();
        
        int maxDay = days.get(0);
        int count = 1;
        
        for (int i = 1; i < days.size(); i++) {
            if (days.get(i) <= maxDay) {
                count++; 
            } else {
                result.add(count);
                maxDay = days.get(i);
                count = 1;
            }
        }
        result.add(count); 
        
        int[] answer = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            answer[i] = result.get(i);
        }

        return answer;
    }
}
